src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class UnconditionalSuppressMessageAttributeState
+	{
+		private readonly AssemblyDefinition _assembly;
+		private GlobalSuppressions _lazyGlobalSuppressions;
+		private readonly Dictionary<MetadataToken, ImmutableDictionary<string, SuppressMessageInfo>> _localSuppressionsByMdToken;
+
+		private bool TryGetTargetScope (SuppressMessageInfo info, out TargetScope scope)
+		{
+			switch (info.Scope) {
+			case "":
+				scope = TargetScope.None;
+				break;
+			case "module":
+				scope = TargetScope.Module;
+				break;
+			case "namespace":
+				scope = TargetScope.Namespace;
+				break;
+			case "resource":
+				scope = TargetScope.Resource;
+				break;
+			case "type":
+				scope = TargetScope.Type;
+				break;
+			case "member":
+				scope = TargetScope.Member;
+				break;
+			case "namespaceanddescendants":
+				scope = TargetScope.NamespaceAndDescendants;
+				break;
+			default:
+				scope = default;
+				return false;
+			}
+
+			return true;
+		}
+
+		private bool HasLocalSuppressions {
+			get {
+				return _localSuppressionsByMdToken.Keys.Count != 0;
```

Comment:
```suggestion
				return _localSuppressionsByMdToken.Count != 0;
```

This should be faster - no need to force allocate the Keys collection